File: 2540-minimum-common-value/2540-minimum-common-value.java
class Solution {
    public int getCommon(int[] nums1, int[] nums2) {
        Set<Integer> first=new HashSet<>();
        Set<Integer> second=new HashSet<>();
        for(int n1:nums1)
        {
            first.add(n1);
        }
        for(int n2:nums2)
        {
            second.add(n2);
        }
       first.retainAll(second);
       int min=Integer.MAX_VALUE;
       for(int val:first)
       {
        if(min>val)
        {
            min=val;
        }
       } 
       if(first.size()==0)return -1;
        return min;
    }
}

File: 2540-minimum-common-value/2540-minimum-common-value.cpp
class Solution {
public:
    int getCommon(vector<int>& nums1, vector<int>& nums2) {
        int k=0;
        int m=0;
        while(k<nums1.size() && m<nums2.size())
        {
            if(nums1[k]==nums2[m])
            {
                return nums1[k];
            }
            else if(nums1[k]>nums2[m])
            {
                m++;
            }
            else
            {
                k++;
            }
        }
        return -1;
    }
};